tests/integration/transformers/test_categorical.py

```diff
@@ -44,6 +45,180 @@ def test_categoricaltransformer_pickle_nans():
+    same as the input.
+
+    Input:
+        - 4 rows of bool data
+    Output:
+        - The reverse transformed data
+    """
+    # setup
+    data = pd.Series([True, False, True, False])
+    transformer = CategoricalTransformer()
+
+    # run
+    reverse = transformer.reverse_transform(transformer.fit_transform(data))
+
+    # assert
+    pd.testing.assert_series_equal(data, reverse)
+
+
+def test_categoricaltransformer_mixed():
+    """Test the CategoricalTransformer on mixed type data.
+
+    Ensure that the CategoricalTransformer can fit, transform, and reverse
+    transform on mixed type data. Expect that the reverse transformed data is
+    the same as the input.
+
+    Input:
+        - 4 rows of mixed data
+    Output:
+        - The reverse transformed data
+    """
+    # setup
+    data = pd.Series([True, 'a', 1, None])
+    transformer = CategoricalTransformer()
+
+    # run
+    reverse = transformer.reverse_transform(transformer.fit_transform(data))
+
+    # assert
+    pd.testing.assert_series_equal(data, reverse)
+
+
+@patch('psutil.virtual_memory')
+def test_categoricaltransformer_mixed_low_virtual_memory(psutil_mock):
+    """Test the CategoricalTransformer on mixed type data with low virtual memory.
+
+    Ensure that the CategoricalTransformer can fit, transform, and reverse
+    transform on mixed type data, when there is low virtual memory. Expect that the
+    reverse transformed data is the same as the input.
+
+    Input:
+        - 4 rows of mixed data
+    Output:
+        - The reverse transformed data
+    """
+    # setup
+    data = pd.Series([True, 'a', 1, None])
+    transformer = CategoricalTransformer()
+
+    virtual_memory = Mock()
+    virtual_memory.available = 1
+    psutil_mock.return_value = virtual_memory
+
+    # run
+    reverse = transformer.reverse_transform(transformer.fit_transform(data))
+
+    # assert
+    pd.testing.assert_series_equal(data, reverse)
+
+
+@patch('psutil.virtual_memory')
+def test_categoricaltransformer_mixed_more_rows(psutil_mock):
+    """Test the CategoricalTransformer on mixed type data with low virtual memory.
+
+    Ensure that the CategoricalTransformer can fit, transform, and reverse
+    transform on mixed type data, when there is low virtual memory and a larger
+    number of rows. Expect that the reverse transformed data is the same as the input.
+
+    Input:
+        - 4 rows of mixed data
+    Output:
+        - The reverse transformed data
+    """
+    # setup
+    data = pd.Series([True, 'a', 1, None])
+    transform_data = pd.Series(['a', 1, None, 'a', True, 1])
+    transformer = CategoricalTransformer()
+
+    virtual_memory = Mock()
+    virtual_memory.available = 1
+    psutil_mock.return_value = virtual_memory
+
+    # run
+    transformer.fit(data)
+    transformed = transformer.transform(transform_data)
+    reverse = transformer.reverse_transform(transformed)
+
+    # assert
+    pd.testing.assert_series_equal(transform_data, reverse)
```

Comment:
@csala I wasn't sure if these two tests should go into integration tests or unit tests -- they are end-to-end but they aren't exactly black box testing, since I am mocking the virtual memory, and am aware of some internal assumptions.